content/00.build/40.tooling/20.hardhat/20.migrating-to-zksync.md

```diff
@@ -1,275 +1,527 @@
+Please use other options like [hardhat-deploy](#deployment-with-hardhat-deploy) or [hardhat-ethers](#deployment-with-hardhat-ethers)
+
+### Deploying proxy contracts
+
+The `@matterlabs/hardhat-zksync` includes `@matterlabs/hardhat-zksync-upgradable`,
+which extends `@openzeppelin/hardhat-upgrades` with all the necessary methods to deploy and upgrade proxy contracts on ZKsync and EVM networks.
+
+The injected `hre.upgrades`
+object provides methods like `deployProxy` so deployment scripts work out of the box.
 
 ::callout{icon="i-heroicons-exclamation-triangle" color="amber"}
-`hardhat-deploy` version `^0.11.26` supports deployments on ZKsync Era.
+To avoid typescript collision errors between `@openzeppelin/hardhat-upgrades` and `@matterlabs/hardhat-zksync-upgradable` make sure that only the latter
+(or the wrapper plugin `@matterlabs/hardhat-zksync`) is imported in the `hardhat.config.ts` file.
 ::
 
-To deploy your contracts you need to use the `Deployer` class from the `hardhat-zksync` plugin.
+See examples below:
 
-Here is a basic deployment script for a `Greeter` contract:
+#### Deployment (Proxy)
 
-```ts
-import { utils, Wallet } from "zksync-ethers";
-import * as ethers from "ethers";
-import { HardhatRuntimeEnvironment } from "hardhat/types";
-import { Deployer } from "@matterlabs/hardhat-zksync";
-
-// An example of a deploy script that will deploy and call a simple contract.
-export default async function (hre: HardhatRuntimeEnvironment) {
-  console.log(`Running deploy script`);
-
-  // Initialize the wallet.
-  const wallet = new Wallet("<WALLET-PRIVATE-KEY>");
-
-  // Create deployer object and load the artifact of the contract we want to deploy.
-  const deployer = new Deployer(hre, wallet);
-  // Load contract
-  const artifact = await deployer.loadArtifact("Greeter");
-
-  // Deploy this contract. The returned object will be of a `Contract` type,
-  // similar to the ones in `ethers`.
-  const greeting = "Hi there!";
-  // `greeting` is an argument for contract constructor.
-  const greeterContract = await deployer.deploy(artifact, [greeting]);
-
-  // Show the contract info.
-  console.log(`${artifact.contractName} was deployed to ${await greeterContract.getAddress()}`);
-}
+::code-group
+
+```typescript [@openzeppelin/contracts-upgradable]
+const constructorArguments = [...];
+const initializerFunctionName = 'initialize'
+const boxFactory = await hre.ethers.getContractFactory("Box");
+const box = await hre.upgrades.deployProxy(boxFactory, constructorArguments, {
+   initializer: initializerFunctionName,
+});
+await box.waitForDeployment();
+```
+
+```typescript [hardhat-zksync-upgradeable]
+const constructorArguments = [...];
+const initializerFunctionName = 'initialize'
+const boxFactory = await hre.ethers.getContractFactory("Box");
+const box = await hre.upgrades.deployProxy(boxFactory, constructorArguments, {
+  initializer: initializerFunctionName
+});
+await box.waitForDeployment();
```

Comment:
Are these supposed to be the exact same?

Reply:
with the new version of the plugins, yes